projects/social/social.py

```diff
@@ -43,12 +49,25 @@ def populateGraph(self, numUsers, avgFriendships):
         # Reset graph
         self.lastID = 0
         self.users = {}
-        self.friendships = {}
-        # !!!! IMPLEMENT ME
+        
+
+        for i in range(numUsers):
+            # generates random names using a syllabary, somewhat unnaturally!
+            randomName = ''
+            for i in range(random.randrange(2,5)):
+                randomName += syllabary[random.randrange(0,len(syllabary))]
+            self.addUser(randomName.capitalize())
 
-        # Add users
+        for user in self.users:
+            for i in range(random.randrange(0,avgFriendships*2+1)):
+                randomFriend = random.randrange(1,len(self.users))
 
```

Comment:
This doesn't seem to be creating the proper amount of friendships.
Consider running a loop the size of the amount of friendships you want (numUsers * avgFriendships // 2),
and then attaching each friendship between 2 random users.